pkg/convert/pprof/streaming/parser_streaming.go

```diff
@@ -361,12 +364,17 @@ func (p *VTStreamingParser) ResolveLabels(l Labels) map[string]string {
 	return m
 }
 
-func (p *VTStreamingParser) buildName(sampleTypeName string, labels map[string]string) *segment.Key {
+func (p *VTStreamingParser) buildName(sampleTypeName string, labels map[string]string) (*segment.Key, error) {
 	for k, v := range p.labels {
 		labels[k] = v
 	}
 	labels["__name__"] += "." + sampleTypeName
-	return segment.NewKey(labels)
+	key := segment.NewKey(labels)
+	err := segment.ValidateKey(key)
+	if err != nil {
+		return nil, err
+	}
+	return key, nil
```

Comment:
One suggestion from my side – please consider moving the validation to [`storage.Put`](https://github.com/pyroscope-io/pyroscope/blob/c79a22fbe459f4c111b338ce8d15a247087243f1/pkg/storage/storage_put.go#L29), or better to [`parser.Ingest`](https://github.com/pyroscope-io/pyroscope/blob/c79a22fbe459f4c111b338ce8d15a247087243f1/pkg/parser/parser.go#L27): this will make it so that the key is validated regardless of the input format. Otherwise we need to add this validation to all the parsers we have

Reply:
Oh, I realised that we actually need to validate keys after [`parser.Ingest`](https://github.com/pyroscope-io/pyroscope/blob/c79a22fbe459f4c111b338ce8d15a247087243f1/pkg/parser/parser.go#L27). So, I would move the validation to [`storage.Put`](https://github.com/pyroscope-io/pyroscope/blob/c79a22fbe459f4c111b338ce8d15a247087243f1/pkg/storage/storage_put.go#L29)